### PR TITLE
added main tests

### PR DESCRIPTION
**Pull request by de-code**

### CODE REVIEW

**Review by jezcope — CHANGES_REQUESTED** [review]
It looks like we have some problems with UTF decoding on Windows right now, which is affecting both this PR and #53. That needs addressing but it's separate to this, so can I suggest marking the tests that fail on Windows currently with:

```python
@pytest.mark.xfail(sys.platform == "win32", reason="Need to fix text decoding on Windows")
```

I'll open another issue to track fixing of the underlying problem in Windows. It's good that we're running into this sooner rather than later. I hate Python text handling!

**Comment by jezcope** [comment]
> ```python
> @pytest.mark.xfail(sys.platform == "win32", reason="Need to fix text decoding on Windows")
> ```
> 
> I'll open another issue to track fixing of the underlying problem in Windows. It's good that we're running into this sooner rather than later. I hate Python text handling!

Unless you can get it fixed, that is!

**Comment by de-code** [comment]
> > ```python
> > @pytest.mark.xfail(sys.platform == "win32", reason="Need to fix text decoding on Windows")
> > ```
> > 
> > 
> > I'll open another issue to track fixing of the underlying problem in Windows. It's good that we're running into this sooner rather than later. I hate Python text handling!
> 
> Unless you can get it fixed, that is!

I am now passing in the encoding and it's default is `utf-8` independent from the platform. Seems to pass now.

File: tests/test__main__.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            '--static',
+            'docs/examples'
+        ])
+
+    def test_should_not_fail_on_runtime_analysis_of_examples(self):
+        main([
+            'docs/examples'
+        ])
+
+    def test_should_raise_error_with_invalid_writer(
```
Comment (jezcope) [inline tests/test__main__.py:53]: This test doesn't look right: it's currently identical to `test_should_not_fail_on_static_analysis_of_examples_including_docstrings` above.

**Review by jezcope — CHANGES_REQUESTED** [review]
I think this looks great, thanks @de-code. The last test in `test__main__.py` doesn't look complete though — is that right?

File: tests/test__main__.py (hunk 1 above)
Comment (de-code) [reply tests/test__main__.py]: You are right, I was going to implement it there but then just tested `parse_args` and forgot to remove the test function.

**Review by jezcope — APPROVED** [review]
Great, some good refactoring and test coverage. Thanks!

**Pull request merged**